parse_date.re

```diff
@@ -908,6 +908,7 @@ year  = [0-9]{1,4};
 year2 = [0-9]{2};
 year4 = [0-9]{4};
 year4withsign = [+-]? [0-9]{4};
+yearx = [+-]? [0-9]{5,19};
```

Comment:
Note: The standard _requires_ the use of a `+` or `-` sign for years greater than four digits. Unless this is specifically intended to be more permissive, that would mean the `?` should be removed and test cases updated accordingly.

Technically, of course, the standard also requires prior agreement between parties on how long the year field should be expanded, mostly to allow the basic format (dates without internal hyphens) to remain unambiguous.  Here, we get around that by simply not accepting expanded representations in the potentially-ambiguous basic format, which I think is reasonable in the name of being flexible in the range of what can be accepted.

Reply:
We are not trying to only parse ISO8601 dates here though, so having it more permissive is fine.

Reply:
_Parsing_ more permissively is fine, but any output should attempt to adhere to the standard by including the sign whenever the year requires more than four digits.

Reply:
I think it would be safer to require it, and we can always relax it later. If ISO 8601 says that this is required, than we can point users to this part of the standard: https://en.wikipedia.org/wiki/ISO_8601#Years